app/javascript/application/coffee_scripts/consultations.coffee

```diff
@@ -77,4 +77,86 @@ $(document).on 'turbolinks:load', ->
+  	parent.addClass('d-none')
+
+	$(document).on 'click', '#add-option-btn', ()->
+		child = $('#options-fields-area').children().last(":nth-child(1)").clone()
+		child.find("input").val("")
+		id = child.data 'id'
+		id = parseInt(id) + 1
+		child.attr("data-id",id)
+		name = "question[sub_questions_attributes][" + id + "][question_text]"
+		child.find("input").attr('name', name)
+		child.children().last().children(":nth-child(1)").removeClass('hidden')
+		$('#options-fields-area').append(child)
+	$(document).on 'click', '.edit-add-option', ()->
+		parent = $(this).parent()
+		if parent.find('select').val() == "checkbox"
+			parent.find('.checkbox-option').removeClass("d-none")
+		else if parent.find('select').val() == "multiple_choice"
+			parent.find('.radio-button-option').removeClass("d-none")
+		else
+			parent.find('.checkbox-option').addClass("d-none")
+			parent.find('.radio-button-option').addClass("d-none")
+			parent.find('.cross-btn').addClass("d-none")
+		child = parent.find('#options-fields-area-edit .checkbox-option-row').last().clone()
+		child = child.removeClass("d-none")
+		child.find(".question_sub_questions__destroy").remove()
+		child.find(".input-box input").val("")
+		id = parent.find('#options-fields-area-edit .checkbox-option-row').length
+		id = parseInt(id) + 1
+		name = "question[sub_questions_attributes][" + id + "][question_text]"
+		child.find(".input-box input").attr('name', name)
+		parent.find('#options-fields-area-edit').append(child)
+	$('#new_question #options-fields-area .checkbox-option-row').addClass("d-none")
+	$('.question_question_type select').each () ->
+		if $(this).val() == "checkbox"
+			$(this).parent().parent().children().find('.radio-button-option').addClass('d-none')
+		else if $(this).val() == "multiple_choice"
+			$(this).parent().parent().children().find('.checkbox-option').addClass('d-none')
+		else
+			$(this).parent().parent().children().find('.radio-button-option').addClass('d-none')
+			$(this).parent().parent().children().find('.checkbox-option').addClass('d-none')
+			$(this).parent().parent().find('.edit-add-option').addClass('d-none')
+			
+	$(document).on 'change', '#new_question #question_question_type', ()->
+		if $(this).val() == "checkbox"
+			$('#add-option-btn').removeClass("d-none")
+			$('.checkbox-option-row').removeClass("d-none")
+			$('.checkbox-option').removeClass("d-none")
+			$('.radio-button-option').addClass("d-none")
+		else if $(this).val() == "multiple_choice"
+			$('#add-option-btn').removeClass("d-none")
+			$('.checkbox-option-row').removeClass("d-none")
+			$('.checkbox-option').addClass("d-none")
+			$('.radio-button-option').removeClass("d-none")
+		else
+			$('#add-option-btn').addClass("d-none")
+			$('.checkbox-option-row').addClass("d-none")
+	$(document).on 'change', '.edit_question .question_question_type select', ()->
+		if $(this).val() == "checkbox"
+			$(this).parent().parent().find('.edit-add-option').removeClass('d-none')
```

Comment:
@BALAJI-BALA - have you considered using `closest` instead of parent.parent?

Reply:
@mikevic Bro the closest will find and return only one div even though if we want many div which has the same class name so in that case I am using find

Reply:
@mikevic sorry bro the closest will find many div with same class_name, but the DIV structure is not in the way to use closest option

Reply:
okay, cool